manual/coding_standards.tex

```diff
@@ -0,0 +1,499 @@
+
+\subsection{Variable Declarations and Definitions}
+
+\begin{itemize}
+\item Avoid declaring multiple variables in the same declaration, especially if they are not fundamental types:
+
+\begin{lstlisting}[showspaces=false]
+int x, y;                        // Not recommended.
+Matrix a("my-matrix"), b(size);  // Disallowed.
+
+// Preferred way.
+int x;
+int y;
+Matrix a("my-matrix");
+Matrix b(10);
+\end{lstlisting}
+
+\item Use the following order for keywords and modifiers in  variable declarations:
+
+\begin{lstlisting}[showspaces=false]
+// General type
+[static] [const/constexpr] Type variable_name;
+
+// Pointer
+[static] [const] Type* [const] variable_name;
+
+// Integer
+// int is optional, if a signedness or size modifier is present.
+[static] [const/constexpr] [signedness] [size] int variable_name;
+
+// Examples:
+static const Matrix a(10);
+const double* const d(3.14);
+constexpr unsigned long l(42);
+\end{lstlisting}
+
+\end{itemize}
+
+\subsection{Function Declarations and Definitions}
+
+The return type should be on the same line as the function name.
+Parameters should be on the same line, too, unless they do not fit on it.
+Include the parameter names also in the declaration of a function, i.e.
+\begin{lstlisting}
+/** calculates a*b+c
+ */
+double function(double a, double b, double c);
+\end{lstlisting}
+avoid
+\begin{lstlisting}
+Type function(Type1, Type2, Type3);
+\end{lstlisting}
+
+\subsection{Conditionals}
+
+Examples:
+\begin{lstlisting}
+if (condition)
+  statement;
+else
+  statement;
+
+if (condition)
+{
+  statement;
+}
+else if (condition2)
+{
+  statement;
+}
+else
+{
+  statement;
+}
+\end{lstlisting}
+
+\subsection{Switch statement}
+
+Switch statements should always have a default case.
+
+Example:
+\begin{lstlisting}
+switch (var) {
+  case 0:
+    statement1;
+    statement2;
+    break;
+
+  case 1:
+    statement1;
+    statement2;
+    break;
+
+  default:
+    statement1;
+    statement2;
+}
+\end{lstlisting}
+
+\subsection{Loops}
+
+Examples:
+\begin{lstlisting}
+for (statement; condition; statement)
+  statement;
+
+for (statement; condition; statement) {
+  statement1;
+  statement2;
+}
+
+while (condition)
+  statement;
+
+while (condition) {
+  statement1;
+  statement2;
+}
+
+do {
+  statement;
+}
+while (condition);
+\end{lstlisting}
+
+\subsection{Preprocessor Directives}
+The preprocessor directives are not indented.
+The hash is the first character of the line.
+
+\subsection{Class Format}
+\label{subsec:class_format}
+\inlinecode{public}, \inlinecode{protected} and \inlinecode{private} keywords are not indented.
+Do not to use \inlinecode{struct} as a way to avoid controlling access to the class. In rare cases where
+a class is a fully public data structure \inlinecode{struct} is appropriate.
+
+Example:
+\begin{lstlisting}
+class Foo : public Bar {
+public:
+  Foo();
+  explicit Foo(int var);
+
+  void function();
+  void emptyFunction() {}
+
+  void set_var(const int var) {
```

Comment:
More curly brace motion needed for these functions.

Reply:
fixed